### PR TITLE
add `origColor` route color when route color is replaced

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
**Description:**
We do a cool thing where we allow overriding the route color. This is useful, but there are cases when we might want the original one anyway, especially within a custom route renderer. This PR allows for this by appending an `origColor` field when the color is overwritten.

**PR Checklist:**
- [X] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [x] Are all languages supported (Internationalization/Localization)?
- [X] Are appropriate Typescript types implemented?

### CODE REVIEW

**Review by daniel-heppner-ibigroup — APPROVED** [review]
Looks good to me, this is a nice place to implement this.

File: lib/actions/apiV2.js [hunk 1]
```diff
@@ -753,8 +765,30 @@ export function findRoutes() {
...
+                    agencyId: route?.agency?.id,
+                    id: route?.route?.id
+                  },
+                  config.transitOperators
+                ),
+                {
+                  color: route?.route?.color,
+                  mode: route.mode
+                }
+              ).split('#')?.[1]
```
Comment (amy-corson-ibigroup) [inline lib/actions/apiV2.js:790]: This method looks almost the same as the one above (and I think there's a third one in this file as well) is there a way to pull this out so you don't have to write it out twice?

**Review by amy-corson-ibigroup — APPROVED** [review]
Looks good! Just had one question

File: lib/actions/apiV2.js (hunk 1 above)
Comment (miles-grant-ibigroup) [reply lib/actions/apiV2.js:790]: Yeah I thought about this but the problem is this is adapted from a core-utils method that's already slightly too wrong to work here so I was thinking you know let's just let this sit for a while, iron out all the changes and make sure that no tiny differences are needed and then we can clean this up in a future refactor!

**Pull request merged**